### PR TITLE
Fix two reasons for flaky tests

### PR DESCRIPTION
**Pull request by karlb**
Structlog messes up capturing with `capsys` in some cases when not properly reset. That was hard to find.

### CODE REVIEW

File: src/raiden_libs/cli.py [hunk 1]
```diff
@@ -100,7 +98,11 @@ def call_with_opened_keystore(**params: Any) -> Callable:
             params["private_key"] = _open_keystore(
                 params.pop("keystore_file"), params.pop("password")
             )
-            return func(**params)
+            try:
```
Comment (palango) [inline src/raiden_libs/cli.py]: Should this go into `call_with_opened_keystore` or a new function?

File: tests/pathfinding/test_api.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def request_path_with(status_code=400, **kwargs):
 
     # bad signature
     bad_iou_dict = good_iou_dict.copy()
-    bad_iou_dict["signature"] = hex(int(bad_iou_dict["signature"], 16) + 1)
+    bad_iou_dict["signature"] = "0x" + "1" * 130
```
Comment (palango) [inline tests/pathfinding/test_api.py]: Why did this start to bite us again?

Comment (karlb) [reply tests/pathfinding/test_api.py]: Because it was used in two different places.

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (karlb) [reply src/raiden_libs/cli.py]: I think this is a good place for it. The name `call_with_opened_keystore` could use some updating, though. Something like `call_with_common_options_initialized`. But that's not a great name, either.

Comment (palango) [reply src/raiden_libs/cli.py]: Not perfect, but better. And I have nothing better to propose...

**Pull request merged**